### ISSUE TITLE
why CLIP use masked self attention rather than self attention

### ISSUE THREAD

**Issue by xiongma** [openai/CLIP#179]

**Comment by jongwook** [comment]
This was explained in section 2.4 of the paper:

> Masked self-attention was used in the text encoder to preserve the ability to initialize with a pre-trained language model or add language modeling as an auxiliary objective, though exploration of this is left as future work.

**Comment by LifeIsStrange** [comment]
@jongwook Masked language modeling significantly reduce performance, see the alternative solution: https://arxiv.org/abs/1906.08237
